src/main/java/io/neonbee/data/DataVerticle.java

```diff
@@ -75,11 +75,206 @@
+        }
+        DeliveryOptions deliveryOptions = deliveryOptions(vertx, null, context);
+        if (context instanceof DataContextImpl) { // will also perform a null check!
+            // remove the verticle right after, as the same context (w/o copying) may be reused for multiple requests
+            ((DataContextImpl) context).popVerticleFromPath();
+        }
+
+        // adapt further delivery options based on the request
+        boolean localOnly = request.isLocalOnly()
+                || (request.isLocalPreferred() && NeonBee.get(vertx).isLocalConsumerAvailable(address));
+        deliveryOptions.setLocalOnly(localOnly);
+        if (request.getSendTimeout() > 0) {
+            deliveryOptions.setSendTimeout(request.getSendTimeout());
+        }
+
+        ofNullable(request.getResolutionStrategy()).map(ResolutionStrategy::name)
+                .ifPresent(value -> deliveryOptions.addHeader(RESOLUTION_STRATEGY_HEADER, value));
+
+        return deliveryOptions;
+    }
+
+    /**
+     * Creates a new delivery options object for any given context.
+     *
+     * @param vertx   the vertx instance
+     * @param codec   the message codec to use (if any)
+     * @param context the data context
+     * @return a new DeliveryOptions
+     */
+    private static DeliveryOptions deliveryOptions(Vertx vertx, MessageCodec<?, ?> codec, DataContext context) {
+        DeliveryOptions deliveryOptions = new DeliveryOptions();
+        deliveryOptions.setSendTimeout(SECONDS.toMillis(NeonBee.get(vertx).getConfig().getEventBusTimeout()))
+                .setCodecName(ofNullable(codec).map(MessageCodec::name).orElse(null));
+        ofNullable(context).map(DataContextImpl::encodeContextToString)
+                .ifPresent(value -> deliveryOptions.addHeader(CONTEXT_HEADER, value));
+        return deliveryOptions;
+    }
+
+    /**
+     * Creates a new data exception for any given throwable cause.
+     *
+     * @param cause any throwable cause
+     * @return a DataException passing the failure code in case it is a ReplyException
+     */
+    private static DataException mapException(Throwable cause) {
+        if (cause instanceof DataException) {
+            return (DataException) cause;
+        }
+
+        int failureCode = FAILURE_CODE_PROCESSING_FAILED;
+        String message = cause.getMessage();
+
+        if (cause instanceof ReplyException) {
+            ReplyException replyException = (ReplyException) cause;
+            switch (replyException.failureType()) {
+            case NO_HANDLERS:
+                failureCode = FAILURE_CODE_NO_HANDLERS;
+                break;
+            case TIMEOUT:
+                failureCode = FAILURE_CODE_TIMEOUT;
+                break;
+            default:
+                failureCode = replyException.failureCode();
+                break;
+            }
+        }
+
+        return new DataException(failureCode, message);
+    }
+
```

Comment:
why was all of this moved? this makesreviewing next to impossible, can you move it back and move it in another change if required?

Reply:
The root cause is that some rule was introduced to enforced the order of the public protected and private methods. I didn't check exactly whether in PMD oder Sonar.

Without moving the code, my PR build always failed.

I don't like it either.